File: pset1/mario/more/mario.c
/*
Program that prints out a double half-pyramid of a specified .

Author: Alejandro Mujica (aledrums@gmail.com)
*/
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // Get pyramid height
    int height = get_int("Height: ");

    while (height < 1 || height > 8)
    {
        height = get_int("Height: ");
    }

    // Generatin each pyramid step
    for (int i = 1; i <= height; ++i)
    {
        // Filling left empty spaces
        for (int j = 0; j < height - i; ++j)
        {
            printf(" ");
        }

        // Drawing left blocks
        for (int j = 0; j < i; ++j)
        {
            printf("#");
        }

        // Middle space
        printf("  ");

        // Drawing right blocks
        for (int j = 0; j < i; ++j)
        {
            printf("#");
        }

        printf("\n");
    }

    return 0;
}

File: pset6/mario/less/mario.py
"""
rogram that prints out a half-pyramid of a specified height.

Author: Alejandro Mujica (aledrums@gmail.com)
"""
import cs50


def main():

    # Getting height
    height = cs50.get_int("Height: ")

    # Reprompting while height is not valid
    while height < 1 or height > 8:
        height = cs50.get_int("Height: ")

    for i in range(1, height + 1):
        # Drawinges empty spaces
        for _ in range(height - i):
            print(' ', end='')

        # Drawing blocks
        for _ in range(i):
            print('#', end='')

        print()


if __name__ == "__main__":
    main()
